packages/trader/src/AppV2/Hooks/useContractsForCompany.ts

```diff
@@ -145,7 +145,10 @@ const useContractsForCompany = () => {
                         available_contract_types[type] = cloneObject(contract_types[type]);
                     }
                     const config: TConfig = available_contract_types[type].config || {};
-                    config.barrier_category = contract.barrier_category as TConfig['barrier_category'];
+                    config.barrier_category =
+                        contract_types[type].barrier_count === 0
+                            ? 'euro_atm'
```

Comment:
if you have any better suggestion please let me know. i dont know how i can extract this info from the contracts_for_company response

Reply:
There isn't, it would be pretty much same solution until `contracts_for_company` is improved

Reply:
I noticed some discrepancies in how the `barrier_category` is assigned to the `rise_fall` and `high_low` contract types based on the data from `contracts_for_company`. Specifically, my expectation was that:
`rise_fall` contract type would be associated with:
- `CALL` with barrier_category `euro_atm`
- `PUT` with barrier_category `euro_atm`

`high_low` contract type would be associated with:
- `CALL` with barrier_category `euro_non_atm`
- `PUT` with barrier_category `euro_non_atm`

However, when I logged the data, I observed the following (please see screenshot below):
`rise_fall`
- `CALL` with barrier_category `euro_atm`
- `CALL` with barrier_category `euro_non_atm`

`high_low`
- `PUT` with barrier_category `euro_atm`
- `PUT` with barrier_category `euro_non_atm`

I suspect the issue may be related to the condition that sets the `type` variable. Please correct me if I'm wrong.

<img width="552" alt="Screenshot 2024-10-01 at 6 04 16 PM" src="https://github.com/user-attachments/assets/6de64962-2630-4874-9988-b7de67f290d8">

Reply:
<img width="407" alt="Screenshot 2024-10-02 at 10 06 16 AM" src="https://github.com/user-attachments/assets/293ee6bc-73f7-423c-95b0-be0c06739bc7">
<img width="391" alt="Screenshot 2024-10-02 at 10 06 11 AM" src="https://github.com/user-attachments/assets/c8c205b6-3f1d-43be-9bed-c1a15f579c2c">
<img width="513" alt="Screenshot 2024-10-02 at 10 06 06 AM" src="https://github.com/user-attachments/assets/3e805558-c474-4946-b935-92b771319504">

Reply:
Huh. This is from `available_contract_types`

<img width="390" alt="Screenshot 2024-10-02 at 11 25 13 AM" src="https://github.com/user-attachments/assets/10a1caa1-299a-403c-8053-bf3fd3ed0bbb">
<img width="405" alt="Screenshot 2024-10-02 at 11 25 20 AM" src="https://github.com/user-attachments/assets/c68c924d-68c2-4978-b810-918a5892ba88">